src/3d/qgs3dsceneexporter.cpp

```diff
@@ -523,7 +525,63 @@ QVector<Qgs3DExportObject *> Qgs3DSceneExporter::processSceneLoaderGeometries( Q
 Qgs3DExportObject *Qgs3DSceneExporter::processGeometryRenderer( Qt3DRender::QGeometryRenderer *geomRenderer, const QString &objectNamePrefix, float sceneScale, QVector3D sceneTranslation )
 {
   // We only export triangles for now
-  if ( geomRenderer->primitiveType() != Qt3DRender::QGeometryRenderer::Triangles ) return nullptr;
+  if ( geomRenderer->primitiveType() != Qt3DRender::QGeometryRenderer::Triangles )
+    return nullptr;
+
+  Qt3DQGeometry *geometry = geomRenderer->geometry();
+
+  // TODO: TO REMOVE
+  qDebug() << "============= Will export geom" << geomRenderer->parent()->objectName();
+
+  // In the case of polygons, we have multiple feature geometries within the same geometry object (QgsTessellatedPolygonGeometry).
+  // The QgsTessellatedPolygonGeometry class holds the list of all feature ids included.
+  // To avoid exporting the same geometry (it can be included in multiple QgsTessellatedPolygonGeometry) more than once,
+  // we keep a list of already exported fid and compare with the fid of the current QgsTessellatedPolygonGeometry.
+  // As we cannot retrieve the specific geometry part for a featureId from the QgsTessellatedPolygonGeometry, we only reject
+  // the geometry if all the featureid are already present.
+  QVector<std::pair<uint, uint>> triangleIndexStartingIndiceToKeep;
+  QgsTessellatedPolygonGeometry *tessGeom = dynamic_cast<QgsTessellatedPolygonGeometry *>( geometry );
+  if ( tessGeom )
+  {
+    QVector<QgsFeatureId> featureIds = tessGeom->featureIds();
+    QVector<uint> triangleIndex = tessGeom->triangleIndexStartingIndices();
+    for ( int idx = 0; idx < featureIds.size(); idx++ )
+    {
+      QgsFeatureId feat = featureIds[idx];
+      // TODO: TO REMOVE
+      qDebug() << "==== Checking feature " << feat ;
+      if ( ! mExportedFeatureIds.contains( feat ) )
+      {
+        // add the feature as it was unknown
+        mExportedFeatureIds.insert( feat );
+
+        // keep the feature triangle indexes
+        uint startIdx = triangleIndex[idx];
+        uint endIdx = idx < triangleIndex.size() - 1 ? triangleIndex[idx + 1] : std::numeric_limits<uint>::max();
+
+        // TODO: TO REMOVE
+        qDebug() << "==== New feature to export '" << feat << "' from idx" << startIdx << "to" << endIdx;
+        triangleIndexStartingIndiceToKeep.append( std::pair<uint, uint>( startIdx, endIdx ) );
+      }
+      else
+      {
+        // keep the feature triangle indexes
+        uint startIdx = triangleIndex[idx];
+        uint endIdx = std::numeric_limits<uint>::max();
+        if ( idx < triangleIndex.size() - 1 )
+          endIdx = triangleIndex[idx + 1];
+        // TODO: TO REMOVE
+        qDebug() << "==== Exclude feature '" << feat << "' from idx" << startIdx << "to" << endIdx;
```

Comment:
This else case is only for debugging, right? Either remove or guard in ifdefs

Reply:
indeed! Removed!